src/PKSim.Core/Services/ParameterTask.cs

```diff
@@ -69,22 +69,34 @@ public interface IParameterTask : ISetParameterTask
       /// </summary>
       /// <param name="parameter">Parameter</param>
       /// <param name="displayUnit">displayUnit</param>
-      ICommand SetParameterDisplayUnit(IParameter parameter, Unit displayUnit);
+      /// <param name="shouldUpdateDefaultStateAndValueOriginForDefaultParameter">
+      ///    If set to <c>true</c> default, default state
+      ///    and value origin of parameter will be updated as well
+      /// </param>
+      ICommand SetParameterDisplayUnit(IParameter parameter, Unit displayUnit, bool shouldUpdateDefaultStateAndValueOriginForDefaultParameter = true);
 
       /// <summary>
       ///    Sets the value in the parameter. Value will be take as is
       /// </summary>
       /// <param name="parameter">Parameter</param>
       /// <param name="value">Value in kernel unit</param>
-      ICommand SetParameterValue(IParameter parameter, double value);
+      /// <param name="shouldUpdateDefaultStateAndValueOriginForDefaultParameter">
+      ///    If set to <c>true</c> default, default state
+      ///    and value origin of parameter will be updated as well
+      /// </param>
+      ICommand SetParameterValue(IParameter parameter, double value, bool shouldUpdateDefaultStateAndValueOriginForDefaultParameter = true);
```

Comment:
It was required to add this extra parameter. True by default no avoid behavior change
But when creating a simulation based on building block, I need to use this flag to avoid updating the IsDefault flag to false when not required